### PR TITLE
DAOS-7903 control: make config generate output consistent with daos_s…

### PR DESCRIPTION
**Pull request by tanabarr**
…erver.yml

dmg config generate command output should only list YAML parameters
documented in daos_server.yml config file.

Test-tag: pr control config_generate_entries

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/cmd/dmg/auto_test.go [hunk 1]
```diff
@@ -97,3 +104,142 @@ func TestDmg_ConfigCommands(t *testing.T) {
 		},
 	})
 }
+
+// TestAuto_ConfigWrite verifies that output from config generate command matches documented
+// parameters in utils/config/daos_server.yml and that private parameters are not displayed.
+// Typical auto-generated output is taken from src/control/lib/control/auto_test.go.
```
Comment (kjacque) [inline src/control/cmd/dmg/auto_test.go:110]: This raises a question for me: Does the output need to be kept in sync with the other test file? If so maybe we should share a text file between the two

**Review by kjacque — COMMENTED** [review]
Overall makes sense. Looks like you need to update one of the engine config unit tests though.

File: src/control/cmd/dmg/auto_test.go (hunk 1 above)
Comment (tanabarr) [reply src/control/cmd/dmg/auto_test.go:110]: the lib/control/auto_test.go:TestControl_AutoConfig_genConfig.hugepages_test,_multiple_target_counts test case doesn't use a test file, the test compares native config structs. sharing a yaml file could work if we change the aforementioned test to decode the expected config from a yaml file. I don't necessarily think they need to be kept in sync, the lib/control/auto_test test is verifying generated config output struct contents while cmd/dmg/auto_test test is verifying encoding of config struct contents to file. A change in one doesn't necessitate a change in the other.

**Review by NiuYawei — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

File: src/control/cmd/dmg/auto_test.go (hunk 1 above)
Comment (kjacque) [reply src/control/cmd/dmg/auto_test.go:110]: Works for me. Thanks.

**Review by kjacque — APPROVED** [review]

**Pull request merged**